### PR TITLE
Now use SourceId

### PR DESCRIPTION
**Pull request by hfiguiere**
Now use SourceId since we have it in glib.

### CODE REVIEW

File: src/signal.rs [hunk 1]
```diff
@@ -41,13 +42,13 @@ fn into_raw<F: FnMut() -> Continue + 'static>(func: F) -> gpointer {
 ///
 /// Similar to `glib::idle_add` but only callable from the main thread and
 /// doesn't require `Send`.
-pub fn idle_add<F>(func: F) -> u32
+pub fn idle_add<F>(func: F) -> SourceId
     where F: FnMut() -> Continue + 'static {
     assert_initialized_main_thread!();
-    unsafe {
+    from_glib(unsafe {
```
Comment (EPashkin) [inline src/signal.rs]: Missed that in glib commit: normally `from_glib` moved to `unsafe` block.

File: src/signal.rs [hunk 2]
```diff
@@ -8,6 +8,7 @@ use std::mem::transmute;
 pub use glib::signal::Inhibit;
 use glib::translate::*;
 
+use glib::SourceId;
```
Comment (EPashkin) [inline src/signal.rs]: Move 2 row upper?

File: src/signal.rs (hunk 1 above)
Comment (hfiguiere) [inline src/signal.rs]: I thought that since the compiler is happy with it, it is better to have it out of the unsafe {} block. Will move it inside.

**Comment by EPashkin** [comment]
:+1: Thanks.

**Comment by GuillaumeGomez** [comment]
Thanks!

**Pull request merged**